[wyongqiing] iP

wyongqiing opened the pull request:
# **NovaBot🪄✨ **  
>*"Your mind is for having ideas, not holding them."* – David Allen  

NovaBot is your intelligent task manager, helping you stay organized effortlessly. It’s:  

- 📝 **Text-based** for simplicity  
- 🚀 **Easy to use** with natural commands  
- ⚡ **Fast and efficient** to manage your tasks  

## **Getting Started**  
1. **Download** NovaBot [here](#).  
2. **Run** the program with a simple double-click.  
3. **Add** your tasks, deadlines, and events.  
4. **Let NovaBot handle your to-dos** for you! 🎯  

## **Features**  
- ✅ **Task management** with to-dos, deadlines, and events  
- 🔍 **Smart search** with improved keyword flexibility  
- 🎙 **Friendly chatbot interface**  
- 📂 **Saves your tasks** so you never lose track  
- 📅 **Upcoming: Reminders & Calendar View**  

## **To-Do List**  
- [x] Implement basic task management  
- [x] Enhance search functionality  
- [x] Add support for marking/unmarking tasks  
- [x] Implement deletion of tasks  
- [ ] Add recurring task support  
- [ ] Implement reminders and notifications  
- [ ] Improve GUI responsiveness  

> **NovaBot ensures you never miss a task again! 🚀**  

```java
// Sample command to add a task
todo Read Java Documentation
deadline Submit report /by 2025-03-01 18:00
event Team meeting /from 2025-03-02 10:00 /to 2025-03-02 12:00 
```

Try **NovaBot** today and boost your productivity! 💡

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Create new chatbot Nova class

Commit:
Add support for echo

Commit:
Add task class

Commit:
Display task list

Commit:
Add task description

Commit:
Add feature to task class

Commit:
Add mark feature

Commit:
add unmark feature

Commit:
Add to do class

Commit:
Add deadline class

Commit:
Add event class

Commit:
Fix typo

Commit:
Move command to TaskManager class

Commit:
Create class with all commands

Commit:
Rearrange the method

Commit:
add exception class

Commit:
Add delete feature

Commit:
Correct typo

Commit:
Remove duke class

Commit:
Test text-ui-test

cyrolite commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,159 @@
+        } catch (NovaException e) {
+            printErrorMessage(e.getMessage());
+        }
+    }
```
Good code readability! The code is very fluent. Good job!

cyrolite commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,159 @@
+        Task task = new Todo(description);
+        tasks.add(task);
+        printTaskAddedMessage(task);
+    }
```
In-built methods inside the class are very descriptive! Good job!

cyrolite commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+public class Task {
+    private String description;
+    private boolean isDone;
```
Good naming convention, with excellent naming of class variables. Good job! LGTM.

Jovin-Ang commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
+
```
Perhaps you could remove this unneeded empty line after the last method. I noticed this in Nova.java too.

Jovin-Ang commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,36 @@
+        taskManager = new TaskManager();
+    }
+
+    public void start() {
+        Scanner sc = new Scanner(System.in);
+
+        // Greeting message
+        taskManager.printGreeting();
+
+        while (true) {
+            String userInput = sc.nextLine();
+
+            if (userInput.equalsIgnoreCase("bye")) {
+                taskManager.printFarewell();
+                break;
+            }
+            // Delegate command handling to the TaskManager
+            taskManager.handleCommand(userInput);
+        }
+
+        sc.close();
+    }
```
I love the use of adding empty lines in between. Makes the code more readable :)

Jovin-Ang commented on src/main/java/NovaException.java:
```diff
@@ -0,0 +1,5 @@
+public class NovaException extends RuntimeException {
+    public NovaException(String message) {
+        super(message);
+    }
+}
```
Should this exception be more specific in what is it used for?

Jovin-Ang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); // X for done, blank for not done
+    }
```
Perhaps this method visibility may be changed to private. I noticed no other outside classes use this method.

Jovin-Ang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); // X for done, blank for not done
+    }
+    @Override
```
opps :sweat_smile: is there a missing empty line above here?

Jovin-Ang reviewed (commented):
Overall, code is clean and easy to read. I noticed the TaskManager.java file is getting a little long, perhaps you can try to separate it into smaller classes.

Commit:
Save task

Commit:
add encode method

Commit:
add save task method

Commit:
Add saveTask method

Commit:
Modify decode method

wyongqiing committed:
Merge pull request #1 from wyongqiing/branch-Level-7

Branch level 7

Commit:
Store date format

Commit:
Update the date of deadline

wyongqiing committed:
Merge pull request #2 from wyongqiing/branch-Level-8

Merge branch level 8

Commit:
Split classes

Commit:
Organise classes into packages

Commit:
Import classes

Commit:
Set up gradle

Commit:
Add JUnit tests

Commit:
Add JavaDoc comments to command classes

Commit:
Add JavaDoc comments to TaskList class

Commit:
Add JavaDoc comments to Parser class

Commit:
Add JavaDoc comments to Storage class

Commit:
Rename variable for readability

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc comments to Storage class
  Add JavaDoc comments to Parser class
  Add JavaDoc comments to TaskList class
  Add JavaDoc comments to command classes

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Rename variable for readability

Commit:
Add find method

Commit:
Modify find command

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Modify find command
  Add find method

wyongqiing committed:
Update README.md

Commit:
add checkstyle file

Commit:
Setting up Java FX

Commit:
Setting up GUI

Commit:
Implement chatbot UI

Commit:
Add executeAndReturn method

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add executeAndReturn method
  Implement chatbot UI
  Setting up GUI
  Setting up Java FX

Commit:
Modify build.gradle file

Commit:
Add assertions to validate critical assumptions

Currently, some key assumptions in task management and command execution
are not explicitly enforced. This could lead to unexpected behavior if
invalid data is passed or unexpected states occur.

Assertions are added to enforce constraints, ensuring non-null values
for descriptions, task lists, and storage, as well as valid indices.

Using assertions documents critical assumptions, helping catch errors
early while keeping the code clean. They are enabled only in debug mode
to avoid performance overhead in production.

Assertions can be enabled using the `-ea` flag when running the program.

Commit:
Delete duplicate class

Commit:
Modify the constant

Commit:
Add Javadoc and enforce Checkstyle compliance

Previously, the code lacked proper Javadoc comments, making it hard
to understand. Additionally, some parts did not conform to the
required coding standards.

To improve maintainability, Javadoc comments were added to all
non-trivial classes and methods. Checkstyle rules were also applied
to ensure consistent formatting.

This change ensures the project adheres to coding standards while
maintaining readability and clarity.

Commit:
Modify check style error

wyongqiing committed:
Merge pull request #4 from wyongqiing/branch-A-Assertions

Branch a assertions

wyongqiing committed:
Merge pull request #5 from wyongqiing/branch-A-CodeQuality

Branch a code quality

Commit:
Modify checkstyle error

Commit:
Improve find feature for partial keyword matching

The find command only returns exact matches, limiting usability.

Users may search with partial keywords, making exact match restrictive.

Modify FindCommand to support substring-based search.

Enhances user experience without altering existing command structure.

Commit:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Improve find feature for partial keyword matching
  Modify checkstyle error

Commit:
Change the width of window

Commit:
Revert "Change the width of window"

This reverts commit 4cfd4261ec82caaf9ee4414c8eb484e771c1813f.

Commit:
Implement help feature

Commit:
Set up a product website

Commit:
Update help feature

Commit:
Add a user guide

Commit:
Modify checkstyle

Commit:
Create a gui package

Commit:
Modify the date parser format

Commit:
Create a jar

Commit:
Delete duplicate classes

Commit:
Upload picture of nova
